ruby_event_store/lib/ruby_event_store/specification.rb

```diff
@@ -0,0 +1,73 @@
+module RubyEventStore
+  class Specification
+    NO_LIMIT = Object.new.freeze
+
+    class Result < Struct.new(:direction, :start, :count, :stream_name)
+      def limit?
+        !count.equal?(NO_LIMIT)
+      end
+
+      def global_stream?
+        stream_name.eql?(GLOBAL_STREAM)
+      end
+
+      def head?
+        start.equal?(:head)
+      end
+
+      def forward?
+        direction.equal?(:forward)
+      end
+
+      def backward?
+        !forward?
```

Comment:
@pawelpacana I would swap the logic so that you check to see if direction is `:backward` and if not, falls to the default of `:forward`... but I do see that `Result` is private, so you're controlling this value.